lib/datadog/appsec/contrib/rack/request.rb

```diff
@@ -9,8 +9,14 @@ def self.query(request)
           end
 
           def self.headers(request)
-            request.each_header.each_with_object({}) do |(k, v), h|
-              h[k.gsub(/^HTTP_/, '').downcase.tr('_', '-')] = v if k =~ /^HTTP_/
+            if request.respond_to?(:each_header)
+              request.each_header.each_with_object({}) do |(k, v), h|
+                h[k.gsub(/^HTTP_/, '').downcase.tr('_', '-')] = v if k =~ /^HTTP_/
+              end
+            else
+              request.env.select { |k, _v| k =~ /^HTTP_/ }.each_with_object({}) do |(k, v), h|
```

Comment:
Minor: This `select` doesn't seem very useful, given we repeat the check anyway in the proc.

Reply:
You, sir, are absolutely correct. I don't know what I've been thinking, adding a useless array allocation just like that!